src/main/java/thaumicboots/item/boots/unimplemented/ItemPurpleVoidwalkerBoots.java

```diff
@@ -1,190 +1,30 @@
-import taintedmagic.common.registry.ItemRegistry;
-import thaumcraft.api.IRepairable;
-import thaumcraft.api.IRunicArmor;
-import thaumcraft.api.IVisDiscountGear;
-import thaumcraft.api.IWarpingGear;
-import thaumcraft.api.aspects.Aspect;
-import thaumcraft.client.fx.ParticleEngine;
-import thaumcraft.client.fx.particles.FXWispEG;
-import thaumcraft.common.Thaumcraft;
-import thaumcraft.common.items.armor.Hover;
-import thaumicboots.main.utils.TabThaumicBoots;
-
-public class ItemPurpleVoidwalkerBoots extends ItemArmor
-        implements IVisDiscountGear, IWarpingGear, IRunicArmor, IRepairable, ISpecialArmor {
+public class ItemPurpleVoidwalkerBoots extends ItemVoidBoots {
 
     public ItemPurpleVoidwalkerBoots(final ArmorMaterial material, final int j, final int k) {
         super(material, j, k);
-        setCreativeTab(TabThaumicBoots.tabThaumicBoots);
-        setUnlocalizedName("ItemPurpleVoidwalkerBoots");
-        setTextureName("thaumicboots:purpleHaze_16x");
-
-        MinecraftForge.EVENT_BUS.register(this);
-    }
-
-    @Override
-    public String getArmorTexture(final ItemStack stack, final Entity entity, final int slot, final String type) {
-        return "thaumicboots:model/VoidwalkerBootsMeteor_-_Purple.png";
-    }
-
-    @Override
-    public EnumRarity getRarity(final ItemStack stack) {
-        return EnumRarity.epic;
-    }
-
-    @Override
-    public int getRunicCharge(final ItemStack stack) {
-        return 0;
-    }
-
-    @Override
-    public int getWarp(final ItemStack stack, final EntityPlayer player) {
-        return 5;
-    }
-
-    @Override
-    public int getVisDiscount(final ItemStack stack, final EntityPlayer player, final Aspect aspect) {
-        return 5;
     }
 
-    @Override
-    public ISpecialArmor.ArmorProperties getProperties(final EntityLivingBase entity, final ItemStack stack,
-            final DamageSource source, final double dmg, final int slot) {
-        int priority = 0;
-        double ratio = damageReduceAmount / 90.0D;
-
-        if (source.isMagicDamage() || source.isFireDamage() || source.isExplosion()) {
-            priority = 1;
-            ratio = damageReduceAmount / 80.0D;
-        } else if (source.isUnblockable()) {
-            priority = 0;
-            ratio = 0.0D;
-        }
-        return new ISpecialArmor.ArmorProperties(priority, ratio, stack.getMaxDamage() + 1 - stack.getItemDamage());
+    protected void setBootsData() {
+        visDiscount = 5;
+        jumpBonus = 0.450D;
+        tier = 3;
+        iconResPath = "thaumicboots:purpleHaze_16x";
+        armorResPath = "thaumicboots:model/VoidwalkerBootsMeteor_-_Purple.png";
+        unlocalisedName = "ItemPurpleVoidwalkerBoots";
     }
 
-    @Override
-    public int getArmorDisplay(final EntityPlayer player, final ItemStack stack, final int slot) {
-        return damageReduceAmount;
-    }
+    // ignore this, it'll only ever matter if this is implemented, leave it be just in case it is
```

Comment:
dead code, remove it. If you ever need to find it again, use git commit history, it's what it is used for.

Reply:
might remove the entire class later tbh, was originally a reference to something Mitch said, but the texture itself was eventually used on a different boot, but anyway, done.